core/Services/Calendar/EventStore.vala

```diff
@@ -330,6 +343,22 @@ public class Calendar.EventStore : Object {
             week_start = 7;
         }
 
+        // Accessing the locale settings
+        if (locale_settings != null) {
+            int week_start_user_pref = locale_settings.get_int ("first-day") + 1;
+
+            if (week_start >= 1 && week_start <= 7) {
+                if (week_start_user_pref == week_start_posix) {
+                    week_start = week_start_posix + glib_offset;
+                } else {
+                    print ("User Preference for week start: %i\nSystem Preference for week start: %i\n", week_start_user_pref, week_start);
+                    week_start = week_start_user_pref + glib_offset;
+                }
+            }
+
+            return (GLib.DateWeekday) week_start;
```

Comment:
This isn't really necessary, since the next line is the same return anyway.

Reply:
Both returns do things differently; one expects the locale setting to exist, and the other one just uses the non-locale setting calculations.

Reply:
Fair enough.